pkg/api/v1/types.go

```diff
@@ -2292,6 +2292,12 @@ type Binding struct {
 	Target ObjectReference `json:"target"`
 }
 
+// Preconditions must be fulfilled before an operation (update, delete, etc.) is carried out.
+type Preconditions struct {
+	// Specifies the target UID.
+	UID *types.UID `json:"uid,omitempty"`
```

Comment:
What does this precondition mean - an object with a given UID exists?
Don't we want to make it slightly more generic - add an ability to specify resourceVersion of this object in which it needs to be?
@caesarxuchao @lavalamp ^^